### ISSUE TITLE
Devices never connect to stf deployment in google cloud

### ISSUE THREAD

**Issue by Any2suited66** [openstf/stf#886]
**What is the issue or idea you have?**
First of all, this piece of software is amazing and the support the devs provide is top notch.  The issue my team and I are experiencing is our devices are recognized by STF but all of them are disconnected.  Our deployment is in google cloud using kubernetes, all 3 devices are on connected to my desktop which is running on MacOS high sierra, and we followed the deployment instructions as best we could for our situation.  We can see the dashboard when we enter our url and the service will often 'disconnect: socket connection lost' and the we have to reconnect.  Also, I entered the adb key in the settings.  Can we get a hint at where to be looking to fix out issue?  Do we need to run adb connect so each device connects to the outside server? Please let me know if you need any more information and I'll be happy to provide it.  Our deployment is based on this [script.](https://github.com/dimasaryo/stf-k8s)

**What is the expected behavior?**
All of our devices are connected and behave gracefully.

**Do you see errors or warnings in the `stf local` output? If so, please paste them or the full log here.**
stf local works fine on my local machine.  that's why we decided to deploy.

**stf doctor output:**
2018-06-08T23:16:18.836Z INF/cli:doctor 84472 [*] OS Arch: x64
2018-06-08T23:16:18.838Z INF/cli:doctor 84472 [*] OS Platform: darwin
2018-06-08T23:16:18.838Z INF/cli:doctor 84472 [*] OS Platform: 17.6.0
2018-06-08T23:16:18.838Z INF/cli:doctor 84472 [*] Using Node 9.4.0
2018-06-08T23:16:18.903Z INF/cli:doctor 84472 [*] Using ZeroMQ 4.2.5
2018-06-08T23:16:18.920Z INF/cli:doctor 84472 [*] Using ADB 1.0.39
2018-06-08T23:16:18.923Z INF/cli:doctor 84472 [*] Using ProtoBuf 3.5.1
2018-06-08T23:16:18.995Z INF/cli:doctor 84472 [*] Using GraphicsMagick 1.3.29
2018-06-08T23:16:19.146Z INF/cli:doctor 84472 [*] Using RethinkDB 2.3.6

**Comment by Any2suited66** [comment]
Here is some output from the logs.  The devices are being ignored.

2018-06-13T15:25:31.688Z INF/provider 128 [*] Found device “04157df438f3b51b” (device)
2018-06-13T15:25:41.708Z INF/provider 128 [*] Providing all 0 of 1 device(s); ignoring “04157df438f3b51b”

**Comment by sorccu** [comment]
The log message is not very good. What it means is that the provider unit was unable to connect to the processor (via triproxy) and receive instructions back. The provider then ignores the devices because it doesn’t know what to do with them. 99% of the time it’s some kind of a connectivity issue.

**Comment by Any2suited66** [comment]
thank you for the quick response.  i will keep looking into and post the solution when we find it.

**Comment by Any2suited66** [comment]
Can you give me a suggestion on how to debug:

> provider unit was unable to connect to the processor (via triproxy) and receive instructions back

thank you for your help.

**Comment by habeebshk** [comment]
Hey did you fix your issue, i have pretty much the same deployment as well, and i am trying it hard to connect.

**Comment by Any2suited66** [comment]
yes, just fixed it today.  so what we found out is deployment cannot be run on macOS by itself since it has connection issues with android just like sorccu said in many of his posts.  our solution was to follow these instructions from [stack overflow](https://stackoverflow.com/questions/29776333/mobile-devices-under-mac-os-x-to-connect-to-docker).  Then deploy with docker using this [project](https://github.com/nikosch86/stf-poc/blob/master/.env).  Worked like a charm.
